lib/charms/lego_base_k8s/v1/lego_client.py

```diff
@@ -0,0 +1,431 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""# lego_client Library.
+
+This library helps user create charms that use LEGO, a letsencrypt client, to provide
+certificates to other charms in the charm ecosystem. The certificate requesting functionality
+and the distribution of issued certificates are all done by the Charm class that's provided
+in this metaclass, which users can inherit to get all of its functionality.
+
+Some configuration by the inheriting charm is required.
```

Comment:
We may want to add a word about LEGO plugins

Reply:
added a small paragraph that explains it